docs/docs/validation-action.mdx

```diff
@@ -0,0 +1,251 @@
+---
+id: validation-action
+sidebar_label: Slot Validation Actions
+title: Slot Validation Actions
+abstract: Learn how `ValidationAction` class is implemented in the Rasa SDK.
+---
+
+There are two helper classes in Rasa SDK with the role of executing custom slot extraction and validation:
+- `ValidationAction`: the base class for custom actions extracting and validating slots that can be set or updated
+outside of a form context.
+- `FormValidationAction`: the base class for custom actions extracting and validating slots that are set only within
+the context of a form.
+
+In order to implement custom slot extraction and validation logic, you have the option of subclassing either
+`ValidationAction` or `FormValidationAction` class, depending on the context in which you want to set or update slots.
+
+## `ValidationAction` class
+
+You can extend the `ValidationAction` class in the Rasa SDK to define custom extraction and / or validation of slots that
+can be set or updated outside of a form context.
+
+:::note
+`ValidationAction` is intended for extracting slots **outside** the context of a form.
+It will ignore extraction and validation methods for any slots that have a form specified under the
+[slot mapping's `conditions`](https://rasa.com/docs/rasa/next/domain#mapping-conditions).
+It will not run these methods when the specified form is active nor when no form is active.
+Please extend the [`FormValidationAction`](./validation-action.mdx#formvalidationaction-class) class to apply
+custom slot mappings only within the context of a form.
+:::
+
+### How to subclass `ValidationAction`
+
+Firstly, you must add the name of this action, `action_validate_slot_mappings`, to the domain `actions` list.
+Note that you do not need to implement the `name` method in the custom action extending `ValidationAction`, since this
+is already implemented. If you override the `name` method, the custom validation action will not run because the
+original name is hardcoded in the call that the default action `action_extract_slots` makes to the action server.
+
+It is recommended to create only one subclass of `ValidationAction` which should contain all extraction and validation
```

Comment:
```suggestion
You should create only one subclass of `ValidationAction` which should contain all extraction and validation
```

Have you checked what happens if you create two? Since there's no point in creating two, this is stronger than a recommendation

Reply:
No, helping out on a different bugfix, not sure I'll be able to test this today.